contacts-app/src/components/ContactForm/ContactForm.jsx

```diff
@@ -14,65 +16,116 @@ class ContactForm extends React.Component {
         companyName: "",
         phone: "",
         email: ""
+      },
+      invalidMessage: {
+        firstName: "",
+        lastName: "",
+        phone: "",
+        email: ""
       }
     };
   }
 
   handleInputChange = event => {
     const update = {};
-    update[event.target.id] = event.target.value;
+    const inputId = event.target.id;
+    const inputValue = event.target.value;
+    if (inputId === "First Name" || "Last Name") {
+      update[inputId] =
+        inputValue.charAt(0).toUpperCase() + inputValue.slice(1);
+    } else {
+      update[inputId] = inputValue;
+    }
     this.setState({
       newContact: { ...this.state.newContact, ...update }
     });
   };
```

Comment:
Its kinda weird here how youre making an object
```
     const update = {};
```
then adding a key to that
```
 update[inputId] = inputValue.charAt(0).toUpperCase() + inputValue.slice(1);
```
then merging that in. 
```
{ ...this.state.newContact, ...update }
```

I think its weird 'cause youre only adding one key. If you were adding multiple things it would probably make more sense. but you could just as easily do:
```
       newContact: { ...this.state.newContact, [inputId]: inputValue }
```
^Though youd still need to do something to format the names. 

Maybe this fn is doing too much?

perhaps something like this could be cleaner: 
```js
  formatName = name => (inputValue.charAt(0).toUpperCase() + inputValue.slice(1));

  handleInputChange = (valueFormatter = i => i ) => event => {
    const inputId = event.target.id;
    const inputValue = event.target.value;
    this.setState({
      newContact: { ...this.state.newContact, [inputId]: valueFormatter(inputValue) }
    });
  };

// ... then when you use it ...

render () => {
  return (
     <Input name="firstName" onChange={this.handleInputChange(this.formatName)} />
     <Input name="phoneNumber" onChange={this.handleInputChange()} />
```

This way you're adding the logic to format the value to the function as a parameter. 
`this.handleInputChange(formatName)` and where theres no formatting required `this.handleInputChange()`. 
> when no formatting is required we have set the default to be a no op func.

Reply:
Great suggestion. I agree that the function is doing too much and extract the format into different function makes it much better.